code/modules/mob/living/carbon/superior_animal/robots/subtype/mining/mining_drone.dm

```diff
@@ -0,0 +1,130 @@
+
+	// Are we going to ores on the ground despite not being supposed to?
+	if(istype(target, /obj/item/ore && !pickup))
+		target = null // reset the target
+
+	if(target) // Do we have a destination?
+		walk_to(src, target, 1, move_to_delay) // Go there
+	else
+		walk_to(src, 0) // Or else stop
+
+/mob/living/carbon/superior_animal/robot/mining/death()
+	drop_loot()
+	..()
+
+/mob/living/carbon/superior_animal/robot/mining/attack_hand(mob/user as mob)
+	if(user.a_intent == I_HELP) // Are we on help intent?
+		var/possible_choices = list("Wandering", "Ore Gathering", "Ore Mining")
+		var/choice = input(user, "Which parameter do you want to toggle in the mining bot?", "Parameters", null) as null|anything in possible_choices
+		switch(choice)
+			if("Wandering")
+				wandering = !wandering
+				to_chat(user, "[src] will [wandering ? "" : "no longer"] wander.")
+			if("Ore Gathering")
+				pickup = !pickup
+				to_chat(user, "[src] will [pickup ? "" : "no longer"] take ore on the ground.")
+			if("Ore Mining")
+				mining = !mining
+				to_chat(user, "[src] will [mining ? "" : "no longer"] mine ore.")
+	else ..()
+
+/mob/living/carbon/superior_animal/robot/mining/attackby(obj/item/W as obj, mob/user as mob)
+	var/obj/item/T // Define the tool variable early on to avoid compilation problem and to allow us to use tool-unique variables
+	if(user.a_intent == I_HELP) // Are we helping ?
+		// If it is a tool, assign it to the tool variable defined earlier.
+		if(istype(W, /obj/item/tool))
+			T = W
+
+		if(QUALITY_PULSING in T.tool_qualities) // We pulse the drone to get the loot
+			drop_loot() // Drop the loot
+			return
+
+	// If nothing was ever triggered, continue as normal
+	..()
+
+/mob/living/carbon/superior_animal/robot/mining/proc/look_around()
+	for(var/O in oview(1, src)) // Check our surroundings.
+		if(mining && istype(O, /turf/simulated/mineral)) // Is it a turf?
+			var/turf/simulated/mineral/M = O
+			if(M.mineral) // Does it have stuff to mine?
+				mine(M) // Mine the turf
+				target = null
+				. = TRUE // We'll return later, keep looping.
+				continue
+
+		if(pickup && istype(O, /obj/item/ore)) // Is it ore on the ground?
+			var/obj/item/ore/Ore = O
+			pick_ore(Ore) // Pick it up
+			target = null
+			. = TRUE // We'll return later, keep looping.
+			continue
+
+	if(.) // Did we mine or pick up stuff earlier on?
+		return . // return TRUE
+
+	for(var/O in oview(viewRange, src)) // Check everything we can see
+		if(pickup && istype(O, /obj/item/ore)) // Is it ore on the ground?
+			var/obj/item/ore/Ore = O
+			target = Ore // Let's go there
+			return TRUE
+
+		if(mining && istype(O, /turf/simulated/mineral)) // Is it a turf?
+			var/turf/simulated/mineral/M = O
+			if(M.mineral) // Does it have stuff to mine?
+				target = M // We want to go to M
+				return TRUE
+
+/mob/living/carbon/superior_animal/robot/mining/proc/mine(var/turf/simulated/mineral/M)
+	visible_message("[src] mine [M]") // Visible message
```

Comment:
should be read as "mines"

Reply:
Nah, because the way it shows is 'Aether Mining Bot mine the rock'

Reply:
(this too is a typo- trust me)